### DISCUSSION TITLE
Edges overlapping over the nodes

### DISCUSSION THREAD

**Discussion by smadapathi** [xyflow/xyflow#2046] [Q&A]
Hello Team,

Thanks for the this awesome library. I have given four edge points around a node(custom one). When I drag/drop an edge from bottom edge point of node to the bottom edge point of another node, edge is drawn on top of the node (refer the pic). Is there any better way to represent the edges in these cases without overlapping over the node?.
<img width="1344" alt="Screenshot 2022-04-05 at 12 42 41 PM" src="https://user-images.githubusercontent.com/2045164/161699308-9ede39c7-4813-442a-bff9-083016935fb1.png">

**Comment by moklick** [comment]
Hey @smadapathi 

this is the expected default behaviour. When you select a node, the node and all connected edges get a higher z-index. If you don't like that you can pass `zIndex:0` as an option to all edges in order to overwrite that behaviour.

**Reply by smadapathi** [reply]
Hi @moklick, thank you for your reply. My question is more around, how we can make the edge not overlapping over the node, like the below pic.
<img width="749" alt="image" src="https://user-images.githubusercontent.com/2045164/161784102-7c621a1e-eb09-4c04-80b8-293d7c9cb764.png">

**Reply by moklick** [reply]
got it ;) We don't have built-in edge routing. You would need to implement a custom edge. Maybe this package can help you to get into the right direction https://github.com/tisoap/react-flow-smart-edge

**Reply by smadapathi** [reply]
Got it @moklick, we will try that out. Thank you again for this library, +1.

**Reply by aShelist** [reply]
> Hi @moklick, thank you for your reply. My question is more around, how we can make the edge not overlapping over the node, like the below pic. <img alt="image" width="749" src="https://user-images.githubusercontent.com/2045164/161784102-7c621a1e-eb09-4c04-80b8-293d7c9cb764.png">

Hello. Have you solved this problem? I would like something like this in my project

**Reply by smadapathi** [reply]
@aShelist, we will be trying out the  https://github.com/tisoap/react-flow-smart-edge in next couple of weeks, will post the updates here..In case, you have any solutions, let us know too..thank you.

**Reply by vaibhav-systango** [reply]
@smadapathi @aShelist @moklick any solutions for this yet ?

**Reply by moklick** [reply]
As explained above a custom edge implementation is the solution here. We won't add this to the React Flow core.

**Reply by vaibhav-systango** [reply]
alright in that case using react flow smart edge seems to be a good alternative @moklick

**Reply by raywei2023** [reply]
cool~

**Comment by awaisshah228** [comment]
I built an open-source package that solves this: [avoid-nodes-edge](https://www.npmjs.com/package/avoid-nodes-edge) (React Flow) and [avoid-nodes-edge-svelte](https://www.npmjs.com/package/avoid-nodes-edge-svelte) (Svelte Flow)

Edges automatically route around nodes with clean orthogonal (90-degree) paths, never crossing through any node. Uses libavoid (C++ WASM) in a Web Worker.

[React Demo](https://avoid-nodes-pro-example.vercel.app/) · [Svelte Demo](https://svelte-demo-beta.vercel.app/) | [npm (React)](https://www.npmjs.com/package/avoid-nodes-edge) · [npm (Svelte)](https://www.npmjs.com/package/avoid-nodes-edge-svelte) | [GitHub](https://github.com/awaisshah228/react-flow-avoid-nodes-routing) | [Package source](https://github.com/awaisshah228/react-flow-avoid-nodes-routing/tree/turbo-package)

React Flow:
npm install avoid-nodes-edge

Svelte Flow:
npm install avoid-nodes-edge-svelte

Make sure to use "libavoid-js": "0.4.5" as a peer dependency.

Works with React Flow v12+ and Svelte Flow (@xyflow/svelte).

**Reply by adamdva** [reply]
Would this work OK with svelte-flow, anyone test it?

**Reply by awaisshah228** [reply]
Yes! I'm the author — I just released a Svelte Flow version as well.

[avoid-nodes-edge-svelte](https://www.npmjs.com/package/avoid-nodes-edge-svelte) — orthogonal edge routing for Svelte Flow. Edges automatically route around nodes with clean 90-degree paths using libavoid (C++ WASM) in a Web Worker.

npm install avoid-nodes-edge-svelte libavoid-js@0.4.5

[Live Demo](https://svelte-demo-beta.vercel.app/)
[npm](https://www.npmjs.com/package/avoid-nodes-edge-svelte)
[GitHub / Source](https://github.com/awaisshah228/react-flow-avoid-nodes-routing/tree/turbo-package/packages/avoid-nodes-edge-svelte)
[Svelte Example ](https://github.com/awaisshah228/react-flow-avoid-nodes-routing/tree/turbo-package/examples-svelte)

Same engine as the React version — group-aware routing, auto best-side detection, configurable spacing/rounding, incremental drag updates, all running off the main thread. Works with @xyflow/svelte.

**Comment by raywei2023** [comment]
[like]  Ray Wei reacted to your message:
________________________________

**Comment by tisoap** [comment]
I'm the creator of [react-flow-smart-edge](https://github.com/tisoap/react-flow-smart-edge) , a library that uses using A* pathfinding to route edges around other nodes, avoiding intersections. It has smart equivalents for every type of edge React Flow has. It's been updated to work with React Flow 12.
